### ISSUE TITLE
False positives in no-dupe-class-members

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#189]
`no-dupe-class-members` does not handle TS overloads correctly:

```
(no-dupe-class-members) Duplicate name 'from'
  static from(
  ~~~~~~~~~~~~
    at /Users/biwanczuk/dev/deno/test_util/../std/node/buffer.ts:57:2

(no-dupe-class-members) Duplicate name 'from'
  static from(buffer: Buffer | Uint8Array): Buffer;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /Users/biwanczuk/dev/deno/test_util/../std/node/buffer.ts:65:2

(no-dupe-class-members) Duplicate name 'from'
  static from(string: string): Buffer;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /Users/biwanczuk/dev/deno/test_util/../std/node/buffer.ts:69:2

(no-dupe-class-members) Duplicate name 'from'
  static from(value: any, offset?: number, length?: number): Buffer {
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /Users/biwanczuk/dev/deno/test_util/../std/node/buffer.ts:71:2

```

Actual source code:
```ts
/**
   * Allocates a new Buffer using an array of bytes in the range 0 – 255. Array
   * entries outside that range will be truncated to fit into it.
   */
  static from(array: number[]): Buffer;
  /**
   * This creates a view of the ArrayBuffer without copying the underlying
   * memory. For example, when passed a reference to the .buffer property of a
   * TypedArray instance, the newly created Buffer will share the same allocated
   * memory as the TypedArray.
   */
  static from(
    arrayBuffer: ArrayBuffer | SharedArrayBuffer,
    byteOffset?: number,
    length?: number
  ): Buffer;
  /**
   * Copies the passed buffer data onto a new Buffer instance.
   */
  static from(buffer: Buffer | Uint8Array): Buffer;
  /**
   * Creates a new Buffer containing string.
   */
  static from(string: string): Buffer;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  static from(value: any, offset?: number, length?: number): Buffer {
    if (typeof value == "string")
      return new Buffer(new TextEncoder().encode(value).buffer);

    // workaround for https://github.com/microsoft/TypeScript/issues/38446
    return new Buffer(value, offset!, length);
  }
```

This rule should probably skip all function that have no body.

CC @magurotuna

**Comment by magurotuna** [comment]
Oh, I should have been careful about overloads. I missed that.
I'll fix the code.
